web3/utils/toolz/__init__.py

```diff
@@ -4,6 +4,7 @@
         complement,
         compose,
         concat,
```

Comment:
Maybe switch to using the `eth_utils.toolz` namespace if you are able to upgrade to a newer `eth-utils`?  Not really part of this PR but just pointing it out.